File: MapA7/MapA7/Repository/IRepository.cs
using MapA7.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MapA7.Repository
{
    public interface IRepository
    {
        void SetFilepath(String FILEPATH);
        void SaveLog(ProgramState ps);
        void Add(ProgramState ps);
        ProgramState GetProgramState(int index);
    }
}

File: MapGUI3/src/Repository/IRepository.java
package Repository;

import Model.ProgramState;

import java.util.List;

public interface IRepository {

    void add(ProgramState ps);
    ProgramState getProgramState(int index);
    void setFilepath(String FILEPATH);
    void logPrgState(ProgramState ps);
    List<ProgramState> getProgramList();
    void setProgramList(List<ProgramState> prgList);
}
